frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -464,20 +489,17 @@ function DatasetEditor(props) {
             small
             onClick={handleCancelClick}
           >{t`Cancel`}</Button>,
-          <ActionButton
-            key="save"
-            disabled={!canSaveChanges}
-            actionFn={handleSave}
-            normalText={dataset.isSaved() ? t`Save changes` : t`Save`}
-            activeText={t`Saving…`}
-            failedText={t`Save failed`}
-            successText={t`Saved`}
-            className={cx(
-              ButtonsS.Button,
-              ButtonsS.ButtonPrimary,
-              ButtonsS.ButtonSmall,
-            )}
-          />,
+          saveButtonTooltipLabel ? (
```

Comment:
Instead of conditionally rendering the `Tooltip` we could pass `disabled={!saveButtonTooltipLabel}` prop to it.
This way we could inline the `ActionButton` back here and remove `key` prop from it.

Reply:
Didn't about about `disabled`, thanks!